### ISSUE TITLE
cryptsetup without key explodifies due to keyscript arg in crypttab

### ISSUE THREAD

**Issue by pdxkay** [brennanfee/linux-bootstraps#65]
I discovered that if one sets `none` as the key file in crypttab, cryptsetup initramfs script barfs and drops you to initramfs busybox.

When removing the keyscript,tries args (`keyscript=/lib/cryptsetup/scripts/passdev,tries=3`) then a normal prompt is restored.

I know that, currently, the script doesn't support setting `none` for a root key--so by deviating from the way the script currently functions regarding crypttab/keying, I brought this on myself. That said, when you get a chance to improve this featureset, this information will be important if you didn't already know.

Thanks.

**Comment by brennanfee** [comment]
I'm not understanding what happened here.  Were these manual edits to crypttab after the script was run?

At present, if you turn encryption on there are three key modes that are supported (using AUTO_DISK_PWD):  the code will generate a keyfile for you (this is the default), you can pass a local file to use as the key file, or you can enter a disk password.

**Comment by pdxkay** [comment]
the tl;dr is this:

`echo "cryptroot UUID=${main_uuid} /dev/disk/by-uuid/${boot_uuid}:root.key luks,initramfs,keyscript=/lib/cryptsetup/scripts/passdev,tries=3${discard_option}" >> /mnt/etc/crypttab` - this is inserted by the script, this does not function unless `root.key` exists (i.e. a password is used) and must be changed to:

`echo "cryptroot UUID=${main_uuid} none luks,initramfs,keyscript=/lib/cryptsetup/scripts/passdev,tries=3${discard_option}" >> /mnt/etc/crypttab` - this change (removal of the hardcoded keyfile) also fails, I'm unsure of the exact reason but removal of keyscript,tries (let initramfs cryptsetup fall back to defaults) with:

`echo "cryptroot UUID=${main_uuid} none luks,initramfs${discard_option}" >> /mnt/etc/crypttab` - functions as expected

**Comment by pdxkay** [comment]
I accidentally hit "close with comment" reopening

**Comment by brennanfee** [comment]
Ok.  This seems to be redundant with #63.  As they are both related to problems when passpharse is used.  I have fixed #63 so this item should be fixed as well.

Closing.
